src/main/java/org/wildfly/security/auth/server/RealmIdentity.java

```diff
@@ -52,9 +52,7 @@
      *
      * @return the decoded principal for this realm identity, or {@code null} if no special decoding is in use
```

Comment:
As long as you're troubleshooting a problem anyway... could you please update this JavaDoc to indicate that all existent RealmIdentitys are required to return a non-null Principal that canonically identifies the identity?